core/src/main/java/org/veriblock/sdk/sqlite/tables/ContainRepository.java

```diff
@@ -8,75 +8,104 @@
 
-    public static final String tableName = "Contain";
-    public static final String txHashColumnName = "txHash";
-    public static final String blockHashColumnName = "blockHash";
+    public static final String tableName = "contain";
+    public static final String txHashColumnName = "tx_hash";
+    public static final String blockHashColumnName = "block_hash";
+    public static final String blockHeightColumnName = "block_height";
+    public static final String blockTimestampColumnName = "block_timestamp";
 
-    public ContainRepository(Connection connection) throws SQLException
-    {
+    public ContainRepository(Connection connection) throws SQLException {
         this.connectionSource = connection;
 
-        Statement stmt = null;
-        try{
-            stmt = connectionSource.createStatement();
+        try(Statement stmt = connectionSource.createStatement()) {
             stmt.execute("CREATE TABLE IF NOT EXISTS " + tableName
                     + "(\n "
                     + txHashColumnName + " TEXT NOT NULL,\n "
                     + blockHashColumnName + " TEXT NOT NULL,\n "
+                    + blockHeightColumnName + " INT NOT NULL,\n "
+                    + blockTimestampColumnName + " INT NOT NULL,\n "
                     + " PRIMARY KEY (" + txHashColumnName + ", " + blockHashColumnName + ")\n "
                     + " FOREIGN KEY (" + txHashColumnName + ")\n "
                     + " REFERENCES " + PoPTransactionsRepository.tableName + " (" + PoPTransactionsRepository.txHashColumnName + ")\n "
                     + ");");
         }
-        finally{
-            if(stmt != null) stmt.close();
-            stmt = null;
-        }
 
-        try {
-            stmt = connectionSource.createStatement();
+        try(Statement stmt = connectionSource.createStatement()) {
             stmt.execute("PRAGMA journal_mode=WAL;");
-        } finally {
-            if(stmt != null) stmt.close();
-            stmt = null;
         }
     }
 
-    public void clear() throws SQLException
-    {
-        Statement stmt = null;
-        try{
-            stmt = connectionSource.createStatement();
+    public void clear() throws SQLException {
+        try(Statement stmt = connectionSource.createStatement()){
             stmt.execute( "DELETE FROM " + tableName);
         }
-        finally {
-            if(stmt != null) stmt.close();
-            stmt = null;
-        }
     }
 
-    public void save(String txHash, String blockHash) throws SQLException
-    {
-        PreparedStatement stmt = null;
-        try {
-            stmt = connectionSource.prepareStatement("REPLACE INTO " + tableName + " ('" + txHashColumnName + "', '" + blockHashColumnName + "') " +
-                    "VALUES(?, ?)");
+    public void save(String txHash, AltChainBlock containingBlock) throws SQLException {
+        String sql = "REPLACE INTO contain (tx_hash, block_hash, block_height, block_timestamp) " +
```

Comment:
Revert this hardcoded data.

Reply:
I don't like solution with arguments concatenation. It looks not clear. ("REPLACE INTO " + tableName + " ('" + txHashColumnName + "', '" + blockHashColumnName + "' .........
We won't change column name in production because we don't have such functionality. (lib for script versioning)

Reply:
You should remove tableName parameter and hardcode it in the SELECT query in such case. I'd prefer to have concatenation instead.

Reply:
use string formatting instead of concatenation
```
String.format("REPLACE INTO {} ({}, {}, {})", tableName, ...);
```